pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -1094,3 +1094,299 @@ func createRandomEcPointHinter(resolver hintReferenceResolver) (hinter.Hinter, e
 
 	return newRandomEcPointHint(p, m, q, s), nil
 }
+
+func newEcRecoverDivModNPackedHint(n, x, s hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "EcRecoverDivModNPacked",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> from starkware.cairo.common.cairo_secp.secp_utils import pack
+			//> from starkware.python.math_utils import div_mod, safe_div
+			//>
+			//> N = pack(ids.n, PRIME)
+			//> x = pack(ids.x, PRIME) % N
+			//> s = pack(ids.s, PRIME) % N
+			//> value = res = div_mod(x, s, N)
+
+			nAddr, err := n.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			xAddr, err := x.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			sAddr, err := s.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			nValues, err := vm.Memory.ResolveAsBigInt3(nAddr)
+			if err != nil {
+				return err
+			}
+
+			xValues, err := vm.Memory.ResolveAsBigInt3(xAddr)
+			if err != nil {
+				return err
+			}
+
+			sValues, err := vm.Memory.ResolveAsBigInt3(sAddr)
+			if err != nil {
+				return err
+			}
+
+			//> N = pack(ids.n, PRIME)
+			nPackedBig, err := secp_utils.SecPPacked(nValues)
+			if err != nil {
+				return err
+			}
+
+			//> x = pack(ids.x, PRIME) % N
+			xPackedBig, err := secp_utils.SecPPacked(xValues)
+			if err != nil {
+				return err
+			}
+			xPackedBig.Mod(&xPackedBig, &nPackedBig)
+
+			//> s = pack(ids.s, PRIME) % N
+			sPackedBig, err := secp_utils.SecPPacked(sValues)
+			if err != nil {
+				return err
+			}
+			sPackedBig.Mod(&sPackedBig, &nPackedBig)
+
+			//> value = res = div_mod(x, s, N)
+			resBig, err := secp_utils.Divmod(&xPackedBig, &sPackedBig, &nPackedBig)
+			if err != nil {
+				return err
+			}
+
+			valueBig := new(big.Int)
+			valueBig.Set(&resBig)
+
+			return ctx.ScopeManager.AssignVariables(map[string]any{"res": &resBig, "value": valueBig})
```

Comment:
i think we only need `res` in scope here, at least given the integration test. But there is no cairo file using this hint in cairolang repo, so its probably more cautious to add both `res` and `value` in scope